src/java.base/share/classes/sun/security/provider/certpath/PKIXCertPathValidator.java

```diff
@@ -241,10 +241,17 @@ private static PKIXCertPathValidatorResult validate(TrustAnchor anchor,
         X509ValidationEvent xve = new X509ValidationEvent();
         if (xve.shouldCommit() || EventHelper.isLoggingSecurity()) {
             int[] certIds = params.certificates().stream()
-                    .mapToInt(x -> x.hashCode())
+                    .mapToInt(Certificate::hashCode)
                     .toArray();
-            int anchorCertId =
-                    anchor.getTrustedCert().hashCode();
+            int anchorCertId = 0;
+            X509Certificate trustedCert = anchor.getTrustedCert();
+            if (trustedCert != null) {
```

Comment:
You could use the `anchorCert` variable which was set earlier instead of calling `getTrustedCert()` again.